### DISCUSSION TITLE
Multiple concurrent appenders for different tables (single database)

### DISCUSSION THREAD

**Discussion by amoghnjoshi** [duckdb/duckdb#4243] [Q&A] [answered]
Hi,

Does Duckdb support using multiple appenders, running concurrently, appending to the different tables from a given database?

**Comment by Mytherin** [comment, marked as answer]
Yes, this is supported, as long as each appender has its own connection instance.

**Reply by Alex-Monahan** [reply]
Small question! Does this work in Python as well?

**Reply by amoghnjoshi** [reply]
Thanks Mytherin, for the quick reply.

However, given the 'single writer or multiple reader' semantics, is it possible to create multiple, write-mode connection instances for same database instance to be used by multiple appenders?

**Reply by Mytherin** [reply]
As long as the connections are in the same process there are no restrictions on the number of writers. The single writer/multiple reader restriction only applies to separate processes using the same database file.

This works in Python as well, but what we call a connection is represented by a cursor in Python.

**Reply by amoghnjoshi** [reply]
Thanks Mytherin, this clears the doubts completely.
